tsl/src/compression/deltadelta.c

```diff
@@ -403,14 +402,16 @@ delta_delta_compressor_append_value(DeltaDeltaCompressor *compressor, int64 next
 /**********************************************************************************/
 
 static void
-int64_decompression_iterator_init_forward(DeltaDeltaDecompressionIterator *iter,
-										  DeltaDeltaCompressed *compressed, Oid element_type)
+int64_decompression_iterator_init_forward(DeltaDeltaDecompressionIterator *iter, void *compressed,
+										  Oid element_type)
 {
-	const char *data = (char *) &compressed->delta_deltas;
-	Simple8bRleSerialized *deltas = bytes_deserialize_simple8b_and_advance(&data);
-	bool has_nulls = compressed->has_nulls == 1;
+	StringInfoData si = { .data = compressed, .len = VARSIZE(compressed) };
+	DeltaDeltaCompressed *header = consumeCompressedData(&si, sizeof(DeltaDeltaCompressed));
+	Simple8bRleSerialized *deltas = bytes_deserialize_simple8b_and_advance(&si);
 
-	Assert(compressed->has_nulls == 0 || compressed->has_nulls == 1);
+	bool has_nulls = header->has_nulls == 1;
+
```

Comment:
Seems to not be used.

Reply:
It's used some lines below. I rewrote another condition to also use it.